DEVELOPER.md

```diff
@@ -56,15 +56,29 @@ Prerequisites:
     * `cargo binstall cargo-deny -y` - for linting dependencies
     * `cargo binstall cargo-udeps -y` - for linting dependencies (detecting unused)
   * To keep all these cargo tools up-to-date use `cargo install-update -a`
+* Database tools:
+  * Install Postgres command line client `psql`: `sudo apt-get install -y postgresql-client`
+  * Install MariaDB command line client `mariadb`: `sudo apt install -y mariadb-client`
+  * Install `sqlx-cli`: `cargo install sqlx-cli --no-default-features --features rustls,postgres,mysql`
 
 Clone the repository:
 ```shell
 git clone git@github.com:kamu-data/kamu-cli.git
+cd kamu-cli
 ```
 
-Build it:
+Setup local databases for `sqlx`:
+```shell
+make sqlx-setup-local
+```
+Note: this step is mandatory for project build to succeed as `sqlx` framework demands database to be accessible at compile-time:
```

Comment:
1) Note that I've explicitly gitignored `.env` files so that a developer has a choice on the preferred method. We can describe the offline mode trick in `DEVELOPER.md`, but every developer will have to choose the method and make a local configuration explicitly. Inserting a default `.env` into Git would rather complicate re-configuration in case it's effect needs to be reverted.

2) I think that at this stage of the project we will be changing tables and queries very frequently, and interfaces of repositories too, so it's not really safe to assume "database stuff is not my business", everyone will have to be aware how it works and what to do with it. What you described in "wants to run DB tests" is actually what I mean, everyone will have to run these tests and be aware how to fix them if anything is wrong, it's unavoidable.

3) >Also, can sqlx run with in-memory only SQLite? Is it considered always "online"?
I'm planning a related experiment to try capabilities in general, I'm curious if we can replace in-memory repository fakes with the SQLite, and whether it makes practical sense. But having this in-memory database "online" is not going to help with the crates that depend on particular database. Remember that query syntax is dialect-specific, types like enums are database-specific, at some point we might have larger differences in the schemas too. The in-memory one will also need to have relevant migrations continously applied.

Reply:
1. I wonder if this can work:
- we add an `.env` file in the root with `SQLX_OFFLINE=true` that we **commit** to git
- when developer runs `make sqlx-local-setup-postgres` other `.env` files are created on lower levels
  - they contain `SQLX_OFFLINE=false` (override) and `DATABASE_URL=xxx`
  - these files are git-ignored
- `make sqlx-clean` would shut down all local databases and clean up all `.env` files except the root - switching you back to the offline mode

Reply:
2. As long as any issue is caught by CI (e.g. updating a query or a DTO without refreshing cache or creating a migration) then I think its fine.

When I'm tweaking something in GQL I don't re-run tests in `kamu-core`. Likewise I'd like any contributor to jump in and compile our project without jumping through too many hoops. If they want to contribute to, for example, CLI usability - I see no reason to force DB knowledge upon them if it can be avoided.

Reply:
Implemented